core/integration/directory_test.go

```diff
@@ -325,38 +325,64 @@ func TestDirectoryWithFile(t *testing.T) {
 }
 
 func TestDirectoryWithoutDirectory(t *testing.T) {
-	t.Parallel()
-
-	dirID := newDirWithFiles(t,
-		"some-file", "some-content",
-		"some-dir/sub-file", "sub-content")
+	// t.Parallel()
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+	defer c.Close()
 
-	var res2 struct {
-		Directory struct {
-			WithDirectory struct {
-				WithoutDirectory struct {
-					Entries []string
-				}
-			}
+	contents := func(s string) dagger.DirectoryWithNewFileOpts {
+		return dagger.DirectoryWithNewFileOpts{
+			Contents: s,
 		}
 	}
 
-	err := testutil.Query(
-		`query Test($src: DirectoryID!) {
-			directory {
-				withDirectory(path: "with-dir", directory: $src) {
-					withoutDirectory(path: "with-dir/some-dir") {
-						entries(path: "with-dir")
-					}
-				}
-			}
-		}`, &res2, &testutil.QueryOptions{
-			Variables: map[string]any{
-				"src": dirID,
-			},
-		})
+	dir1 := c.Directory().
+		WithNewFile("some-file", contents("some-content")).
+		WithNewFile("some-dir/sub-file", contents("sub-content"))
+
+	entries, err := dir1.
+		WithoutDirectory("some-dir").
+		Entries(ctx)
+
+	require.NoError(t, err)
+	require.Equal(t, []string{"some-file"}, entries)
+
+	dir := c.Directory().
+		WithNewFile("foo.txt", contents("foo")).
+		WithNewFile("a/bar.txt", contents("bar")).
+		WithNewFile("a/data.json", contents("{\"datum\": 10}")).
+		WithNewFile("b/foo.txt", contents("foo")).
+		WithNewFile("b/bar.txt", contents("bar")).
+		WithNewFile("b/data.json", contents("{\"datum\": 10}")).
+		WithNewFile("c/file-a1.txt", contents("file-a1.txt")).
+		WithNewFile("c/file-a1.json", contents("file-a1.json")).
+		WithNewFile("c/file-b1.txt", contents("file-b1.txt")).
+		WithNewFile("c/file-b1.json", contents("file-b1.json"))
+
+	entries, err = dir.Entries(ctx)
+	require.NoError(t, err)
+	require.Equal(t, []string{"a", "b", "c", "foo.txt"}, entries)
+
+	entries, err = dir.
+		WithoutDirectory("a").
+		Entries(ctx)
+	require.NoError(t, err)
+	require.Equal(t, []string{"b", "c", "foo.txt"}, entries)
+
+	entries, err = dir.
+		WithoutDirectory("b/*.txt").
```

Comment:
> Feels a little weird for this to be calling `WithoutDirectory` but with a glob that refers to files. Since the fix is made to the shared `Without` call (and I think that's a good thing), I think this can just call `WithoutFile`.

I like this suggestion. I'll do it. 

> We actually don't have a test here showing that directories themselves are globbable, so maybe we could add that and turn this into one big `TestDirectoryWithoutDirectoryWithoutFIle` and remove the test below?

Sounds good! I'll add the changes, and ping you guys again for a review!